src/Servers/Kestrel/Core/src/Internal/Http/HttpResponsePipeWriter.cs

```diff
@@ -35,7 +37,7 @@ public override void CancelPendingFlush()
         public override void Complete(Exception exception = null)
         {
             ValidateState();
-            _pipeControl.Complete(exception);
+            _completeTask = _pipeControl.CompleteAsync(exception);
```

Comment:
This sets us up for an inconsistent / confusing end-to-end story. Let's go over that before continuing here.
https://github.com/aspnet/AspNetCore/issues/7370#issuecomment-506806184